packages/evm/contracts/registry/GuardiansRegistry.sol

```diff
@@ -207,6 +219,44 @@ contract GuardiansRegistry is ControlledRecoverable, IGuardiansRegistry, ERC900,
         _stake(_from, _from, _amount, _data);
     }
 
+    /**
+    * @notice Lock `@tokenAmount(self.token(), _amount)` of sender's active balance
+    * @param _lockManager Address of the lock manager that will control the lock
+    * @param _amount Amount of active tokens to be locked
+    */
+    function lockActivation(address _lockManager, uint256 _amount) external {
+        _lockActivation(msg.sender, _lockManager, _amount);
+    }
+
+    /**
+    * @notice Unlock `@tokenAmount(self.token(), _amount)` of `_guardian`'s active balance
+    * @param _guardian Address of the guardian unlocking the active balance of
+    * @param _lockManager Address of the lock manager controlling the lock
+    * @param _amount Amount of active tokens to be unlocked
+    * @param _deactivate Whether the requested amount must be deactivated too
+    */
+    function unlockActivation(address _guardian, address _lockManager, uint256 _amount, bool _deactivate) external {
+        bool canUnlock = _lockManager == msg.sender || ILockManager(_lockManager).canUnlock(_guardian, _amount);
+        require(canUnlock, ERROR_CANNOT_UNLOCK_ACTIVATION);
+
+        ActivationLocks storage activationLocks = guardiansByAddress[_guardian].activationLocks;
+        uint256 lockedAmount = activationLocks.lockedBy[_lockManager];
+        require(lockedAmount > 0, ERROR_ZERO_LOCK_ACTIVATION);
+        require(_amount > 0, ERROR_UNLOCK_ACTIVATION_ZERO);
+
+        uint256 unlockedAmount = lockedAmount > _amount ? _amount : lockedAmount;
+        uint256 newLockedAmount = lockedAmount.sub(unlockedAmount);
+        uint256 newTotalLocked = activationLocks.total.sub(unlockedAmount);
+
+        activationLocks.total = newTotalLocked;
+        activationLocks.lockedBy[_lockManager] = newLockedAmount;
+        emit GuardianActivationLockDecreased(_guardian, _lockManager, newLockedAmount, newTotalLocked);
+
+        if (msg.sender == _guardian && _deactivate) {
```

Comment:
Hmm, would it be a problem if the lock manager wanted to do this?

Reply:
In that case the lock manager couldn't simply answer `true` to `canUnlock` so the guardian has complete control on his tokens?

Reply:
agreed offline to avoid having the lock managers allowed to deactivate tokens